docs/page/adding-new-language.md

```diff
@@ -11,24 +11,39 @@ corresponding mode -\> language id - in this case `(python-mode .
 
 
 ``` elisp
-(defvar lsp-language-id-configuration
-  '(...
-    (python-mode . "python")
-    ...))
-;; if you are adding the support for your language server in separate repo use
-;; (add-to-list 'lsp-language-id-configuration '(python-mode . "python"))
-
-(lsp-register-client
- (make-lsp-client :new-connection (lsp-stdio-connection "pyls")
-                  :activation-fn (lsp-activate-on "python")
-                  :server-id 'pyls))
+(with-eval-after-load 'lsp-mode
+  (defvar lsp-language-id-configuration
+    '(...
+      (python-mode . "python")
+      ...))
+  ;; if you are adding the support for your language server in separate repo use
+  ;; (add-to-list 'lsp-language-id-configuration '(python-mode . "python"))
+
+  (lsp-register-client
+    (make-lsp-client :new-connection (lsp-stdio-connection "pyls")
+                     :activation-fn (lsp-activate-on "python")
+                     :server-id 'pyls)))
 ```
 
 `lsp-mode` is using `lsp-language-id-configuration` to determine what is the
 buffer language. When the `major-mode` is not sufficient to determine the
 language (e.g. `web-mode` is used for `javascript`, `html`, and `css`) you can put
 regex.
 
+Here's an example of how to specify a filetype along with web-mode. In this case it's Shopify `.liquid` files.
+
+```elisp
+;; Shopify template lsp with theme-check
+(with-eval-after-load 'lsp-mode
+  (add-to-list 'lsp-language-id-configuration
+    '(web-mode . "\\.liquid\\'"))
```

Comment:
In this example, "\\.liquid\\'" should be a language like `"java"`, `"c"`. Also, selecting web-mode as an example is not appropriate since web-mode can be used for multiple languages.

Reply:
This specific example is an edge-case for Shopify's own flavor of liquid template syntax. There is no solid, pre-existing program mode for it. This is why web-mode is used in this example. It is able to highlight Shopify liquid syntax with only a few minor issues. All I have to go on is the extension for this case.

I am open to better ways to do this aside from writing my own "shopify-mode" package. That's something I'm wanting to do in the long term. For everyone else right now, I'm not sure what to suggest.

Reply:
I have refined both my config and this example to use a derived major mode. It's important to note that the documentation already stated that regexp is an acceptable means of setting up a custom language server, but I do see the benefit in being explicit with your own major major mode. I included a link to some more documentation to help new users.

Commit f086163ff1e87a2a3d6058a0aca70ae32d88fe10

Reply:
I hope these updates have been satisfactory 🙂. Please let me know if there's more I may do